### PR TITLE
Support redirects after Ajax transitions

### PR DESCRIPTION
**Pull request by ramonski**
## Description of the issue/feature this PR addresses

This PR supports automatic redirects after Ajax transitions (introduced in #108 and #110) by providing custom (named) multi adapters for the object/transition.

## Current behavior before PR

Automatic redirects, e.g. auto-print stickers for Samples or auto-partitioning from Sample templates was not possible

## Desired behavior after PR is merged

Automatic redirects, e.g. auto-print stickers for Samples or auto-partitioning from Sample templates is possible by providing a custom multi-adapter:

```xml
    <!-- Custom listing adapter for sample transition "receive" -->
    <adapter
        for="senaite.app.listing.interfaces.IAjaxListingView
            bika.lims.interfaces.IAnalysisRequest
            senaite.core.interfaces.ISenaiteCore"
        factory=".workflow.SampleReceiveWorkflowTransition"
        name="receive"
        />
    </configure>
```

--
I confirm I have tested the PR thoroughly and coded it according to [PEP8][1]
standards.

[1]: https://www.python.org/dev/peps/pep-0008

### CODE REVIEW

File: src/senaite/app/listing/adapters/workflow.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        """Execute the workflow transition
+        """
+        obj = self.context
+        oid = api.get_id(obj)
+
+        try:
+            obj = api.do_transition_for(obj, transition)
+        except ConflictError:
+            self.error["message"] = _(
+                "A database conflict occured during transition "
```
Comment (xispa) [inline src/senaite/app/listing/adapters/workflow.py]: typo: *occured* --> *occurred*

I would also rephrase as *A database conflict error occurred during ...*

File: src/senaite/app/listing/adapters/workflow.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            # NOTE: We do not propagate back to the UI when the transition
+            #       failed, because it is most of the time an expected
+            #       side-effect. E.g. when an analysis with calculation
+            #       dependencies is submitted, the dependent analyses are
+            #       submitted as well. Therefore, if the current object is
+            #       such a dependency, it might fail here.
+            logger.warn(exc)
+        except Exception as exc:
+            self.error["message"] = _(
+                "An unkown error occured during transition '{}' on '{}': {}"
```
Comment (xispa) [inline src/senaite/app/listing/adapters/workflow.py]: *occured* --> *occurred*

File: src/senaite/app/listing/ajax.py [hunk 3]
```diff
@@ -633,33 +632,38 @@ def ajax_do_action_for(self):
 
         uids = payload.get("uids")
         transition = payload.get("transition")
+
         errors = {}
+        redirects = {}
 
         for uid in uids:
+            import pdb; pdb.set_trace()
```
Comment (xispa) [inline src/senaite/app/listing/ajax.py]: remove this pdb

File: webpack/app/listing.coffee [hunk 4]
```diff
@@ -1919,10 +1927,19 @@ class ListingController extends React.Component
   on_api_error: (response) ->
     @toggle_loader off
     console.debug "°°° ListingController::on_api_error: GOT AN ERROR RESPONSE: ", response
-    response.text().then (data) =>
-      title = _t("Oops, an error occured! 🙈")
-      message = _t("The server responded with the status #{response.status}: #{response.statusText}")
+
+    title = _t("Oops, an error occured! 🙈")
```
Comment (xispa) [inline webpack/app/listing.coffee]: *occured* --> *occurred*

**Review by xispa — CHANGES_REQUESTED** [review]
Some typos and a pdb

File: src/senaite/app/listing/ajax.py (hunk 3 above)
Comment (ramonski) [reply src/senaite/app/listing/ajax.py]: Fixed in 7e31540

File: src/senaite/app/listing/adapters/workflow.py (hunk 1 above)
Comment (ramonski) [reply src/senaite/app/listing/adapters/workflow.py]: Fixed in 11af065

**Review by xispa — APPROVED** [review]

**Pull request merged**